cpp/arcticdb/version/version_store_api.cpp

```diff
@@ -583,7 +582,10 @@ VersionedItem PythonVersionStore::write_versioned_composite_data(
     }
 
     auto frames = create_input_tensor_frames(sub_keys, items, norm_metas, user_metas, cfg().write_options().empty_types());
+    // Need to hold the GIL up to this point as we will call pb_from_python
```

Comment:
`create_input_tensor_frames` only depends on the input parameters, so we can release the GIL right at the start if we move this call up above line 568